### PR TITLE
harmonize scoring based on extension (fixes #109)

### LINKED ISSUE #109: Score COPYING over COPYING.foo files
Would need to update the scoring in https://github.com/benbalter/licensee/blob/master/lib/licensee/project_files/license_file.rb#L18.

### PR DESCRIPTION
**Pull request by waldyrious**
This change implements the following ranking:

1. No extension: LICENSE, COPYING, etc.
3. Filetype extension (syntactic annotation): UNLICENSE.txt, COPYRIGHT.md, etc.
2. Content-related extension (semantic annotation): COPYING.image, LICENCE.go, etc.

### CODE REVIEW

File: lib/licensee/project_files/license_file.rb [hunk 1]
```diff
@@ -14,10 +14,13 @@ def attribution
 
       def self.name_score(filename)
         return 1.0 if filename =~ /\A(un)?licen[sc]e\z/i
-        return 0.9 if filename =~ /\A(un)?licen[sc]e\.(md|markdown|txt)\z/i
-        return 0.8 if filename =~ /\Acopy(ing|right)(\.[^.]+)?\z/i
-        return 0.7 if filename =~ /\A(un)?licen[sc]e\.[^.]+\z/i
-        return 0.5 if filename =~ /licen[sc]e/i
+        return 0.9 if filename =~ /\Acopy(ing|right)\z/i
```
Comment (benbalter) [inline lib/licensee/project_files/license_file.rb]: I think `LICENSE.md` should still take precedence over `COPYING` when both our present (but agree `COPYING` should take precedence over `LICENSE.image`)

**Comment by benbalter** [comment]
@waldyrious This is great. One comment on the priorities, and then we'll just need to update the corresponding tests.

File: lib/licensee/project_files/license_file.rb (hunk 1 above)
Comment (waldyrious) [inline lib/licensee/project_files/license_file.rb]: Sure, I'll change that.

**Comment by waldyrious** [comment]
Shouldn't the tests be relative (score of form X > score of form Y) rather than binded to absolute values?

**Comment by benbalter** [comment]
> Shouldn't the tests be relative (score of form X > score of form Y) rather than binded to absolute values?

If there's a concise way to write that, that would better match our intent here, yes, I believe the actual values can be treated as arbitrary.

**Comment by waldyrious** [comment]
fixed the tests and rebased the branch.

**Comment by benbalter** [comment]
Thanks @waldyrious! I'll get those rubocop errors clean up on `master` as I believe I may have introduced at least one of them. 😄

**Comment by waldyrious** [comment]
Thanks :)

**Pull request merged**